packages/flutter_tools/lib/src/build_system/targets/dart.dart

```diff
@@ -140,6 +124,28 @@ class CopyFlutterBundle extends Target {
   ];
 }
 
+/// A helper function to copy an [assetBundle] into an [environment]'s output directory,
+/// plus an optional [pathSuffix]
+Future<void> copyAssets(AssetBundle assetBundle, Environment environment, [String pathSuffix = '']) async {
+  final Pool pool = Pool(kMaxOpenFiles);
```

Comment:
If the goal is to limit the number of simultaneous file operations by the `flutter_tools` process, we should have one global `Pool` object used by everything (perhaps stored inside the `AppContext`). Perhaps it could even be built into `fs`.

Reply:
That sounds like a good enhancement here, but I'm a bit worried about last minute changes. I'll open a separate PR to do that with more testing